src/lotus/client.rs

```diff
@@ -491,6 +537,100 @@ impl<T: JsonRpcClient + Send + Sync> LotusClient for LotusJsonRPCClient<T> {
     }
 }
 
+impl<T: JsonRpcClient + Send + Sync> LotusJsonRPCClient<T> {
+    fn sign_mpool_message(&self, msg: &MpoolPushMessage) -> anyhow::Result<Signature> {
+        if self.wallet_store.is_none() {
+            return Err(anyhow!("key store not set, function not supported"));
+        }
+
+        let message = fvm_shared::message::Message {
```

Comment:
Out of a curiosity, is this how forest does it? By creating an `fvm::Message` and signing its CID?

Reply:
yeah, I copied from them. They are actually using `message.cid()` with `Impl Cbor ...`. But `Cbor` trait seems to be deprecated. So I just copied the implementation over.

Reply:
Cool!